azurerm/resource_arm_private_link_service_test.go

```diff
@@ -0,0 +1,342 @@
+					resource.TestCheckResourceAttr(resourceName, "tags.env", "test"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMPrivateLinkServiceExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Private Link Service not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		client := testAccProvider.Meta().(*ArmClient).network.PrivateLinkServiceClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		if resp, err := client.Get(ctx, resourceGroup, name, ""); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Private Link Service %q (Resource Group %q) does not exist", name, resourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on network.PrivateLinkServiceClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMPrivateLinkServiceDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).network.PrivateLinkServiceClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_private_link_service" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, name, ""); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on network.PrivateLinkServiceClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMPrivateLinkService_basic(rInt int, location string) string {
+	standardResources := testAccAzureRMPrivateLinkServiceTemplate_standardResources(rInt, location)
+	privateLink := testAccAzureRMPrivateLinkServiceTemplate_basic(rInt)
+
+	return testAccAzureRMPrivateLinkServiceTemplate("", standardResources, privateLink)
+}
+
+func testAccAzureRMPrivateLinkService_update(rInt int, location string) string {
+	standardResources := testAccAzureRMPrivateLinkServiceTemplate_standardResources(rInt, location)
+	privateLink := testAccAzureRMPrivateLinkServiceTemplate_update(rInt)
+
+	return testAccAzureRMPrivateLinkServiceTemplate("", standardResources, privateLink)
+}
+
+func testAccAzureRMPrivateLinkService_complete(rInt int, location string) string {
+	subscriptionDataSource := testAccAzureRMPrivateLinkServiceTemplate_subscriptionDataSource()
+	standardResources := testAccAzureRMPrivateLinkServiceTemplate_standardResources(rInt, location)
+	privateLink := testAccAzureRMPrivateLinkServiceTemplate_complete(rInt)
+
+	return testAccAzureRMPrivateLinkServiceTemplate(subscriptionDataSource, standardResources, privateLink)
+}
```

Comment:
I don't see the value in this, we could just include `subscriptionDataSource` bit in the `standardResources` and call it a day

Reply:
fixed.